packages/nitro-protocol/contracts/AssetHolder.sol

```diff
@@ -312,15 +331,153 @@ contract AssetHolder is IAssetHolder {
             } else {
                 holdings[allocation[m].destination] += payoutAmount;
             }
-            // Event emitted regardless of success of external calls
+            // Event emitted 
             emit AssetTransferred(channelId, allocation[m].destination, payoutAmount);
         }
     }
 
+    /**
+     * @notice Transfers as many funds escrowed against `guarantorChannelId` as can be afforded for a specific destination in the beneficiaries of the __target__ of that channel.  Does not check allocationBytes or guarantee against on chain storage.
+     * @dev Transfers as many funds escrowed against `guarantorChannelId` as can be afforded for a specific destination in the beneficiaries of the __target__ of that channel.  Does not check allocationBytes or guarantee against on chain storage.
+     * @param guarantorChannelId Unique identifier for a guarantor state channel.
+     * @param guarantee The guarantee
+     * @param allocationBytes The abi.encode of AssetOutcome.Allocation for the __target__
+     * @param destination External destination or channel to transfer funds *to*.
+     */
+    function _claim(
+        bytes32 guarantorChannelId,
+        Outcome.Guarantee memory guarantee,
+        bytes memory allocationBytes,
+        bytes32 destination
+    ) internal {
+
+        Outcome.AllocationItem[] memory allocation = abi.decode(
+        allocationBytes,
+        (Outcome.AllocationItem[])
+        );
+        uint256 balance = holdings[guarantorChannelId];
+        uint256 affordsForDestination;
+        uint256 residualAllocationAmount;
+        uint256 _amount;
+        uint256 j; // indexes guarantee 
+        uint256 i; // indexes target allocations
+
+        for (j = 0; j < guarantee.destinations.length; j++) {
+            // for each _destination in the guarantee
+            bytes32 _destination = guarantee.destinations[i];
+            // if destination matches the one that is passed in
+            // loop over allocations in the target channel and decrease balance until we hit the specified destination
+            if (_destination == destination) {
```

Comment:
I'm confused whether this does the right thing? It seems like it should be:

* For each _destination in guarantee:
   * Iterate over the allocation to find _amt it sends to that destination
   * If _destination != destination
        * Decrease balance by _amt
        * Next _destination
    * Else
        * affords = min(balance, _amt)
        * break

whereas it seems like once you find the destination in the guarantee, you then walk along the allocation as normal decreasing the balance?

Reply:
Yeah you are correct. I even put a bad test case in that passed with this behaviour. 

Hopefully fixed in a6cac4b